### ISSUE TITLE
feat: use TypeScript's emit SDK

### ISSUE THREAD

**Issue by ffMathy** [ftrackhq/ftrack-ts-schema-generator#8]
Right now, we emit strings for our TypeScript.

That's great, but there's a more accurate and better way of doing it via TypeScript's compiler API.
https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API

That way, we can't accidentally make incorrect TypeScript. It also makes it easier to infer things about the project (such as whether or not the Ftrack API package has been installed etc).

**Comment by gismya** [comment]
I'm not familiar with TypeScript's compiler API. What section of that documentation would be applicable to this?

**Comment by ffMathy** [comment]
It's actually in the link above. But more specifically it'd be this section:
https://github.com/microsoft/TypeScript/wiki/Using-the-Compiler-API#creating-and-printing-a-typescript-ast

Or a better more simple third party guide:
https://shadeglare.medium.com/typescript-code-generation-using-its-compiler-api-4c50ad9f7884

**Comment by gismya** [comment]
Not a bad idea, but I don't think the benefits outweigh the time to rewrite (Even though it looks really fun). But if an extensive rewrite is done in the future, it's definitely a way to go.

**Comment by ffMathy** [comment]
Fair enough. Let's leave the project open though. I might do it at some point when I'm bored in a weekend or something.

**Comment by ffMathy** [comment]
Let's close this for now. I think you might be right that the added benefit of this is low, especially when using eslint.
